### PR TITLE
[+]Locality LB user guide

### PR DESCRIPTION
**Pull request by derekwin**

### CODE REVIEW

File: content/en/docs/userguide/locality_loadbalance.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+```
+&workloadapi.LoadBalancing{
+	// Prefer endpoints in close zones, but allow spilling over to further endpoints where required.
+	RoutingPreference: []workloadapi.LoadBalancing_Scope{
+		workloadapi.LoadBalancing_NETWORK,
+		workloadapi.LoadBalancing_REGION,
+		workloadapi.LoadBalancing_ZONE,
+		workloadapi.LoadBalancing_SUBZONE,
+	},
+	Mode: workloadapi.LoadBalancing_FAILOVER,
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:36]: we donot need to show the underlying details. Only need to document how user can setup the service to use this feature. 

And to be more clear, list all options and the expected results

File: content/en/docs/userguide/locality_loadbalance.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+```
+  - With annotation [todo: need to verify the feasibility.]
+```
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    networking.istio.io/traffic-distribution: PreferClose
+```
+todo: 没有在istio的api.annotation找到traffic-distribution这个标签，但是istio的server.go显示了通过这个字段配置策略的可能性。
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:58]: It is there istio.io/api/annotation/annotations.gen.go

```
	NetworkingTrafficDistribution = Instance {
		Name:          "networking.istio.io/traffic-distribution",
		Description:   `Controls how traffic is distributed across the set of available endpoints.

At this time, this annotation only impacts routing done by Ztunnel.

Accepted values:
* "PreferClose": endpoints will be categorized by how "close" they are, consider network, region, zone, and subzone.
  Traffic will be prioritized to the closest healthy endpoints.
  For example, if I have a Service with "PreferClose" set, with endpoints in zones "us-west,us-west,us-east". When 
  sending traffic from a client in zone "us-west", all traffic will go to the two "us-west" backends.
  If one those backends become unhealthy, all traffic will go to the remaining endpoint in "us-west".
  If that backend becomes unhealthy, traffic will sent to "us-east".
`,
```

File: content/en/docs/userguide/locality_loadbalance.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+2. Local: A strict mode that only matches the current NODE.
+```
+&workloadapi.LoadBalancing{
+  // Only allow endpoints on the same node.
+  RoutingPreference: []workloadapi.LoadBalancing_Scope{
+    workloadapi.LoadBalancing_NODE,
+  },
+  Mode: workloadapi.LoadBalancing_STRICT,
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:67]: This canbe deleted, only need to mention the visible results with strict setting

File: content/en/docs/userguide/locality_loadbalance.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+      nodeSelector:
+        kubernetes.io/hostname: ambient-worker
+EOF
+```
+
+- test
+```
+kubectl exec "$(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}')" -c sleep -- curl -sSL "http://helloworld:5000/hello"
+```
+The output is from the helloworld-region.zone1.subzone1 that is currently co-located on the ambient-worker.
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:242]: We need more tests, like: when a local endpoints down. It automatically failover to the remote one.

**Comment by LiZhenCheng9527** [comment]
If this PR is not ready to be merged in, you can add the `/hold` tag

**Comment by derekwin** [comment]
> If this PR is not ready to be merged in, you can add the `/hold` tag

I think it is ready now.

File: content/en/docs/userguide/locality_loadbalance.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+      nodeSelector:
+        kubernetes.io/hostname: ambient-worker
+EOF
+```
+
+- Test the access.
+```
+kubectl exec "$(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}')" -c sleep -- curl -sSL "http://helloworld:5000/hello"
+```
+The output is from the helloworld-region.zone1.subzone1 that is currently co-located on the ambient-worker.
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:267]: How can we verify the res is from `elloworld-region.zone1.subzone1`

File: content/en/docs/userguide/locality_loadbalance.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+        kubernetes.io/hostname: ambient-worker3
+EOF
+```
+
+test multi times:
+```
+kubectl exec "$(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}')" -c sleep -- curl -sSL "http://helloworld:5000/hello"
+```
+
+The output randomly shows helloworld-region.zone1.subzone2 and helloworld-region.zone1.subzone2-worker3.
```
Comment (hzxuzhonghu) [inline content/en/docs/userguide/locality_loadbalance.md:333]: same, you should paste the output as a proof

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

Some nits can be fixed later

**Pull request merged**